File: apps/query-burst/src/main/java/com/booster/queryburst/order/domain/OrderStatus.java
package com.booster.queryburst.order.domain;

/**
 * 주문 상태
 * - 상태별 분포: DELIVERED(50%) > SHIPPED(20%) > PAID(15%) > CANCELED(10%) > PENDING(5%)
 * - 상태 필터링은 카디널리티가 낮아 단독 인덱스보다 복합 인덱스가 효과적
 */
public enum OrderStatus {
    PENDING,    // 결제 대기
    PAID,       // 결제 완료
    SHIPPED,    // 배송 중
    DELIVERED,  // 배송 완료
    CANCELED    // 취소
}

File: apps/kotlin/shopping-service/src/main/kotlin/com/booster/kotlin/shoppingservice/order/domain/OrderStatus.kt
package com.booster.kotlin.shoppingservice.order.domain

enum class OrderStatus {
    CREATED,
    PAYMENT_PENDING,
    PAID,
    PREPARING,
    SHIPPED,
    DELIVERED,
    PAYMENT_FAILED,
    CANCELED,
    REFUND_REQUESTED,
    REFUNDED;

    fun canCancel() = this in setOf(CREATED, PAYMENT_PENDING, PAID)
    fun canRequestRefund() = this == DELIVERED
}
